pkg/sql/pgwire/types.go

```diff
@@ -538,11 +536,26 @@ func writeBinaryBytes(b *writeBuffer, v []byte, t *types.T) {
 		v = []byte{0}
 	}
 
-	if paddingNeeded(v, t) {
-		b.writePaddedData(v, t)
-	} else {
-		b.writeLengthPrefixedByteSlice(v, int32(len(v)))
+	pad := paddingForCharType(len(v), t)
+	b.putInt32(int32(len(v) + pad))
+	b.write(v)
+
+	// apply padding (in the form of blanks spaces) to the right of the write buffer
+	for pad > 0 {
+		n := min(pad, len(spaces))
+		b.write(spaces[:n])
+		pad -= n
+	}
+}
+
+// paddingForCharType checks if we have a blankPaddedCharType and if the input length is less
+// than the specified length to determine if padding is needed. If padding is needed we return
+// the difference between the specified length and the input length else we return 0.
+func paddingForCharType(inputLength int, t *types.T) int {
```

Comment:
nit: `bpcharPadding` or `bpcharPad`

Reply:
Great I just changed it to bpcharPadding, thanks @mgartner!
<!-- Reviewable comment -ODNaB-k-ACPW1gghAhq:b-p3ru4a -->
<!-- Sent from Reviewable.io -->